src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -856,8 +856,10 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
             }
         });
 
-        let ast_attrs: &[ast::Attribute] = &self.cx.tcx.get_attrs(item.def_id);
-        let inner_docs = ast_attrs.inner_docs();
+        let inner_docs = match self_id {
+            Some(did) => self.cx.tcx.get_attrs(did).inner_docs(),
+            None => false,
+        };
```

Comment:
This is wrong. It will break if this is documenting a function and not a type. You should check item.is_fake instead.

Reply:
```rust
let self_id = if item.is_fake() {
            None
        // Checking if the item is a field in an enum variant
        }
```
`self_id` is already the result of testing `item.is_fake()`. How is using it different than checking `item.is_fake()` again ?

Reply:
`tcx.parent` can return `None`. And this is not really the check you want anyway, the self ID is irrelevant when what you actually care about is whether the item is fake.

Reply:
Ok. It's done